### PR TITLE
#672: Add OpenAPI endpoint for Admin Resources

### PR DESCRIPTION
**Pull request by gdasbach**
#672: Add additional OpenAPI endpoint for documenting available admin endpoints

### CODE REVIEW

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import io.swagger.v3.oas.integration.SwaggerConfiguration;
+import io.swagger.v3.oas.models.OpenAPI;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.servlet.ServletConfig;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.*;
```
Comment (johngmyers) [inline openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Import the Proofpoint code style and reformat. Wildcard imports are not permitted.

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @Operation(hidden = true)
+    public Response getOpenApiSpec(
+            @Context HttpHeaders headers,
+            @Context UriInfo uriInfo,
+            @PathParam("type") String type,
+            @Context ServletConfig config,
+            @Context Application app)
+            throws Exception
+    {
+        if(getOpenApiConfiguration() == null){
```
Comment (johngmyers) [inline openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Space after if.

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            SwaggerConfiguration oasConfig = new SwaggerConfiguration()
+                    .openAPI(new OpenAPI())
+                    .prettyPrint(true);
+            oasConfig.setResourceClasses(app.getSingletons().stream().map(classame -> classame.getClass().getName()).collect(Collectors.toSet()));
```
Comment (johngmyers) [inline openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Break this stream into separate lines.

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            oasConfig.setScannerClass(SWAGGER_JAXRS_ANNOTATION_SCANNER);
+            setOpenApiConfiguration(oasConfig);
+        }
+
+        Response response = getOpenApi(headers, config, app, uriInfo, type);
+        response.getHeaders().add("Access-Control-Allow-Origin", "*");
+        return response;
+    }
+}
```
Comment (johngmyers) [inline openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: missing newline at end of file.

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java (hunk 1 above)
Comment (gdasbach) [reply openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Done

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java (hunk 2 above)
Comment (gdasbach) [reply openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Done

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java (hunk 3 above)
Comment (gdasbach) [reply openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Done

File: openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java (hunk 4 above)
Comment (gdasbach) [reply openapi/src/main/java/com/proofpoint/openapi/OpenApiAdminResource.java]: Done

**Comment by johngmyers** [comment]
Rebased, squashed, and merged.

**Pull request closed without merging**